File: experiments/kittentts.py
# pip install https://github.com/KittenML/KittenTTS/releases/download/0.8.1/kittentts-0.8.1-py3-none-any.whl
# pip install soundfile

from kittentts import KittenTTS
import soundfile as sf

# Models: KittenML/kitten-tts-nano-0.8-fp32 (15M), kitten-tts-micro-0.8 (40M), kitten-tts-mini-0.8 (80M)
# Voices: Bella, Jasper, Luna, Bruno, Rosie, Hugo, Kiki, Leo
model = KittenTTS("KittenML/kitten-tts-nano-0.8-fp32")

text = "Hello, this is a test of Kitten TTS running without a GPU."
audio = model.generate(text, voice="Luna", speed=1.0, clean_text=True)

out = "kittentts-out.wav"
sf.write(out, audio, 24000)
print(f"Saved {out}")

File: experiments/kittentts.ts
import { pipeline } from "@huggingface/transformers";
import { WaveFile } from "wavefile";
import Speaker from "speaker";

const VOICES = ["Bella", "Jasper", "Luna", "Bruno", "Rosie", "Hugo", "Kiki", "Leo"];

async function main() {
  console.log("Loading KittenTTS Nano v0.8...");
  const tts = await pipeline("text-to-speech", "onnx-community/kitten-tts-nano-0.1-ONNX");

  const voice = "Luna";
  const text = "Hello, this is Kitten TTS version 0.8 running in Node.js.";

  console.log(`Generating audio with voice: ${voice}`);
  const out = await tts(text, { voice });

  const wav = new WaveFile();
  wav.fromScratch(1, out.sampling_rate, "32f", out.audio);

  const speaker = new Speaker({
    channels: 1,
    bitDepth: 32,
    sampleRate: out.sampling_rate,
    // @ts-ignore
    float: true,
  });
  speaker.write(wav.toBuffer());

  console.log(`Done — sample rate: ${out.sampling_rate} Hz, samples: ${out.audio.length}`);
}

main();
